From f3158328981e668a24848750aacb22bde72f3bcc Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Mon, 23 Oct 2017 16:53:52 +0800
Subject: [PATCH 2740/5242] MLK-16682 usb: cdns3: gadget: delete useless
 is_iso_flag

commit  157f65cd5ab4347f1db55dae7878e8c6033b3ac9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since the runtime endpoint type is decided by device descriptors,
we delete useless is_iso_flag which is decided during the initialization.
It also fixed a bug the max_packet_size is determined wrongly for
high/full speed connection.

BuildInfo:
- SCFW 8dcff26, IMX-MKIMAGE ea027c4b, ATF
- U-Boot 2017.03-imx_v2017.03_4.9.51_imx8_beta1+g6dc7b0f

Acked-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    8 +++-----
 drivers/usb/cdns3/gadget.h |    1 -
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index 673d3b9..dd6984b 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -1357,13 +1357,13 @@ static void cdns_ep_config(struct usb_ss_endpoint *usb_ss_ep)
 		break;
 
 	case USB_SPEED_FULL:
-		max_packet_size = (usb_ss_ep->is_iso_flag ?
+		max_packet_size = (is_iso_ep ?
 			ENDPOINT_MAX_PACKET_SIZE_1023 :
 			ENDPOINT_MAX_PACKET_SIZE_64);
 		break;
 
 	case USB_SPEED_HIGH:
-		max_packet_size = (usb_ss_ep->is_iso_flag ?
+		max_packet_size = (is_iso_ep ?
 			ENDPOINT_MAX_PACKET_SIZE_1024 :
 			ENDPOINT_MAX_PACKET_SIZE_512);
 		break;
@@ -1976,10 +1976,8 @@ static int usb_ss_init_ep(struct usb_ss_dev *usb_ss)
 			usb_ss_ep->endpoint.caps.dir_out = 1;
 
 		/* check endpoint type */
-		if (iso_ep_reg & (1uL << ep_reg_pos)) {
+		if (iso_ep_reg & (1uL << ep_reg_pos))
 			usb_ss_ep->endpoint.caps.type_iso = 1;
-			usb_ss_ep->is_iso_flag = 1;
-		}
 
 		if (bulk_ep_reg & (1uL << ep_reg_pos)) {
 			usb_ss_ep->endpoint.caps.type_bulk = 1;
diff --git a/drivers/usb/cdns3/gadget.h b/drivers/usb/cdns3/gadget.h
index 136fd6e..a1363cd 100644
--- a/drivers/usb/cdns3/gadget.h
+++ b/drivers/usb/cdns3/gadget.h
@@ -168,7 +168,6 @@ struct usb_ss_endpoint {
 	char name[20];
 	int hw_pending_flag;
 	int stalled_flag;
-	int is_iso_flag;
 	int wedge_flag;
 	void *cpu_addr;
 	dma_addr_t dma_addr;
-- 
1.7.9.5

